### PR TITLE
Unit test EmojiRepository

### PR DESCRIPTION
**Pull request by Scott-Caldwell**
Adds unit tests for `EmojiRepository`.

I ran into a snag while trying to test the methods that use `FromSql`. Apparently the in-memory database doesn't support that. (https://github.com/aspnet/EntityFrameworkCore/issues/7212)

```
System.NotSupportedException: 'Could not parse expression 'value(Microsoft.EntityFrameworkCore.Query.Internal.EntityQueryable`1[Modix.Data.Models.Emoji.SingleEmojiStatsDto]).AsNoTracking().FromSql(value(Microsoft.EntityFrameworkCore.RawSqlString), __p_0)': This overload of the method 'Microsoft.EntityFrameworkCore.RelationalQueryableExtensions.FromSql' is currently not supported.'
```

I'll take a look into it soon and see if anyone's come up with a solution to aid in unit testing, but for now I think this is a good start.

### CODE REVIEW

**Comment by JakenVeina** [comment]
The entire way `.FromSql()` works is that it lets you write queries directly against the underlying database, which is PostgreSQL, Any testing against those queries is impossible without some kind of PostgreSQL database to test against. Or, at least something that can parse the PostgreSQL flavor of SQL. That's basically the primary reason we should avoid such queries as much as possible.

For now, we basically can't test those without running the full app.

File: Modix.Data.Test/Repositories/EmojiRepositoryTests.cs [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+        #region CreateAsync() Tests
+
+        [Test]
+        public async Task CreateAsync_DataIsNull_DoesNotUpdateEmojisAndThrowsException()
+        {
+            (var modixContext, var uut) = BuildTestContext();
+
+            await Should.ThrowAsync<ArgumentNullException>(uut.CreateAsync(null));
+
+            modixContext.Emoji.Select(x => x.Id).ShouldBe(Emoji.Entities.Select(x => x.Id));
```
Comment (JakenVeina) [inline Modix.Data.Test/Repositories/EmojiRepositoryTests.cs]: For the sake of clarity and idiot-proffing, you should specify the optional parameter `ignoreOrder: true` for all the `.ShouldBe()` calls like this.

We could also take it a step further an make a `.ShouldBeSetEqual()` extension method, or something similar.

File: Modix.Data.Test/Repositories/EmojiRepositoryTests.cs [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+        [TestCaseSource(nameof(EmojiCreationTestCases))]
+        public async Task CreateMultipleAsync_DataIsNotNull_InsertsEmojis(EmojiCreationData data)
+        {
+            const int RecordCount = 100;
+
+            (var modixContext, var uut) = BuildTestContext();
+
+            await uut.CreateMultipleAsync(data, RecordCount);
+
+            var emojis = modixContext.Emoji
```
Comment (JakenVeina) [inline Modix.Data.Test/Repositories/EmojiRepositoryTests.cs]: Could we instead have the method return the ID's it generated, like the `CreateAsync()` method does?

File: Modix.Data.Test/Repositories/EmojiRepositoryTests.cs [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+        }
+
+        [TestCaseSource(nameof(NonexistentDeleteTestCases))]
+        public async Task DeleteAsync_Nonexistent_DoesNotDeleteEmojis(EmojiSearchCriteria criteria)
+        {
+            (var modixContext, var uut) = BuildTestContext();
+
+            await uut.DeleteAsync(criteria);
+
+            modixContext.Emoji.Count().ShouldBe(Emoji.Entities.Count());
```
Comment (JakenVeina) [inline Modix.Data.Test/Repositories/EmojiRepositoryTests.cs]: if we're checking that the set of IDs matches, this is redundant.

File: Modix.Data.Test/Repositories/EmojiRepositoryTests.cs [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+
+            await uut.DeleteAsync(criteria);
+
+            modixContext.Emoji.Count().ShouldBe(Emoji.Entities.Count() - deletedCount);
+            modixContext.Emoji.EachShould(x => x.ShouldNotHaveChanged());
+        }
+
+        #endregion DeleteAsync() Tests
+
+        #region Single GetEmojiStatsAsync() Tests
```
Comment (JakenVeina) [inline Modix.Data.Test/Repositories/EmojiRepositoryTests.cs]: Nit/FYI

Usually, when working with overloads, I qualify them by listing what the param types or names are (at least, the important ones). 

For this, I would probably do `GetEmojiStatsAsync(emoji)` with `GetEmojiStatsAsync_Emoji_` and `GetEmojiStatsAsync(emojiIds)` with `GetEmojiStatsAsync_EmojiIds`.

**Comment by Scott-Caldwell** [comment]
Closing until I have time to work on this

**Pull request closed without merging**